biorxiv/corpora_comparison/corpora_comparison.py

```diff
@@ -0,0 +1,668 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Comparative Linguistic Analysis of bioRxiv and PMC
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+from collections import defaultdict, Counter
+import csv
+import itertools
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import pickle
+import spacy
+from scipy.stats import chi2_contingency
+from tqdm import tqdm_notebook
+
+from annorxiver_modules.corpora_comparison_helper import(
+    aggregate_word_counts,
+    dump_to_dataframe,
+    get_term_statistics,
+    KL_divergence
+)
+
+
+# # Full Text Comparison (Global)
+
+# ## Gather Word Frequencies
+
+# In[2]:
+
+
+biorxiv_count_path = Path("output/total_word_counts/biorxiv_total_count.tsv")
+pmc_count_path = Path("output/total_word_counts/pmc_total_count.tsv")
+nytac_count_path = Path("output/total_word_counts/nytac_total_count.tsv")
+
+
+# In[3]:
+
+
+if not biorxiv_count_path.exists():
+    biorxiv_corpus_count = (
+        aggregate_word_counts(
+            list(Path("output/biorxiv_word_counts").rglob("*tsv"))
+        )
+    )
+    dump_to_dataframe(biorxiv_corpus_count, "output/biorxiv_total_count.tsv")
+    biorxiv_corpus_count.most_common(10)
+
+
+# In[4]:
+
+
+if not pmc_count_path.exists():
+    pmc_corpus_count = (
+        aggregate_word_counts(
+            list(Path("../../pmc/pmc_corpus/pmc_word_counts").rglob("*tsv"))
+        )
+    )
+    dump_to_dataframe(pmc_corpus_count, "output/pmc_total_count.tsv")
+    pmc_corpus_count.most_common(10)
+
+
+# In[5]:
+
+
+if not nytac_count_path.exists():
+    nytac_corpus_count = (
+        aggregate_word_counts(
+            list(Path("../../nytac/corpora_stats/output").rglob("*tsv"))
+        )
+    )
+    dump_to_dataframe(nytac_corpus_count, "output/nytac_total_count.tsv")
+    nytac_corpus_count.most_common(10)
+
+
+# In[6]:
+
+
+biorxiv_total_count_df = pd.read_csv(
+    biorxiv_count_path.resolve(), 
+    sep="\t"
+)
+
+pmc_total_count_df = pd.read_csv(
+    pmc_count_path.resolve(), 
+    sep="\t"
+)
+
+nytac_total_count_df = pd.read_csv(
+    nytac_count_path.resolve(), 
+    sep="\t"
+)
+
+
+# In[7]:
+
+
+biorxiv_sentence_length = pickle.load(
+    open("output/biorxiv_sentence_length.pkl", "rb")
+)
+pmc_sentence_length = pickle.load(
+    open("../../pmc/pmc_corpus/pmc_sentence_length.pkl", "rb")
+)
+nytac_sentence_length = pickle.load(
+    open("../../nytac/corpora_stats/nytac_sentence_length.pkl", "rb")
+)
+
+
+# In[8]:
+
+
+spacy_nlp = spacy.load('en_core_web_sm')
+stop_word_list = list(spacy_nlp.Defaults.stop_words)
+
+
+# ## Get Corpora Comparison Stats
+
+# In[9]:
+
+
+biorxiv_sentence_len_list = list(biorxiv_sentence_length.items())
+biorxiv_data = {
```

Comment:
>Not something you should do for this PR and I'm not sure what your timeline looks like moving forward. But if you think it makes sense to refactor, you could pull out as a function.

Hmmm.. I see what you are saying. I think I'll put that on the refactor list rather than change it as I don't think the function would be needed anywhere else in my repository.